components/base/sensorcontrolled/movestraight.go

```diff
@@ -0,0 +1,191 @@
+// Package sensorcontrolled base implements a base with feedback control from a movement sensor
+package sensorcontrolled
+
+import (
+	"context"
+	"math"
+	"time"
+
+	geo "github.com/kellydunn/golang-geo"
+
+	"go.viam.com/rdk/components/movementsensor"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+const (
+	slowDownDistGain      = .1
+	maxSlowDownDist       = 100 // mm
+	moveStraightErrTarget = 20  // mm
+	headingGain           = 1.
+)
+
+func (sb *sensorBase) MoveStraight(
+	ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{},
+) error {
+	ctx, done := sb.opMgr.New(ctx)
+	defer done()
+	sb.setPolling(false)
+	straightTimeEst := time.Duration(int(time.Second) * int(math.Abs(float64(distanceMm)/mmPerSec)))
+	startTime := time.Now()
+	timeOut := 5 * straightTimeEst
+	if timeOut < 10*time.Second {
+		timeOut = 10 * time.Second
+	}
+
+	if sb.position == nil || len(sb.conf.ControlParameters) == 0 {
+		sb.logger.CWarnf(ctx,
+			"Position reporting sensor not available, and no control loop is configured, using base %s MoveStraight",
+			sb.controlledBase.Name().ShortName())
+		sb.stopLoop()
+		return sb.controlledBase.MoveStraight(ctx, distanceMm, mmPerSec, extra)
+	}
+
+	initialHeading, err := sb.headingFunc(ctx)
+	if err != nil {
+		return err
+	}
+	// make sure the control loop is enabled
+	if sb.loop == nil {
+		if err := sb.startControlLoop(); err != nil {
```

Comment:
I think you also need to also check here that the control Parameters or the control Config exists.

Reply:
kk I missed that you can have control parameters and no loop config set, if the velocities are not set on the base. I had a check on `len(sb.conf.ControlParameters)` but I'll switch that to `sb.controlLoopConfig`

Reply:
i misread the reconfigure code so my above comment isn't true. I do not use controls if `len(sb.conf.ControlParameters) == 0`

Reply:
switched the check to `len(sb.controlLoopConfig.Blocks)`